autograder-api/src/main/java/de/firemage/autograder/api/CodePosition.java

```diff
@@ -0,0 +1,15 @@
+package de.firemage.autograder.api;
+
+import java.nio.file.Path;
+
+public interface CodePosition {
+    Path path();
+    int startLine();
+
+    int endLine();
+
+    int startColumn();
+
+    int endColumn();
+    String readCompilationUnit();
```

Comment:
It is difficult to understand what this does without a javadoc. How about `readSourceFile`?